tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,214 @@
+    }
+    die "Failed to $power_action the guest." if (!esxi_vm_power_getstate($vm_id) =~ m/$expect_vm_state/);
+}
+
+sub wait_for_vm_bootup {
+    my $vm_ip       = shift;         # vm ip address
+    my $if_pingable = shift;         # 1 or 0
+    my $times       = shift // 6;    # the number of ping execution
+    my $count       = shift // 3;    # ping waits for the count
+    my $interval    = shift // 5;    # wait secs between each ping
+    my $cmd;
+
+    if ($if_pingable) {
+        $cmd = "ping -c$count -i$interval $vm_ip";
+    } else {
+        $cmd = "! ping -c$count -i$interval $vm_ip";
+    }
+
+    while ($times && console('svirt')->run_cmd($cmd)) {
+        $times--;
+    }
+}
+
+sub login_vm_console {
+    reset_consoles;
+    select_console('sut');
+    assert_screen('grub2', 90);
+    send_key 'ret';
+    select_console('root-console', timeout => 180);
+}
+
+sub set_guest_time {
+    my $n_hrs  = (sprintf "%02d", rand(24));
+    my $n_min  = (sprintf "%02d", rand(59));
+    my $n_sec  = (sprintf "%02d", rand(59));
+    my $g_time = script_output("date -u -s $n_hrs:$n_min:$n_sec +'\%F \%T'");
+
+    return $g_time;
+}
+
+sub get_timelc {
+    my ($time, $timelc, $sec, $min, $hour, $mday, $mon, $year);
+    $time = $_[0];
+
+    my @time = reverse(split /-|:|\s|\//, $time);
+    foreach (@time) {
+        s/^0//;
+        $timelc .= $_ . ",";
+    }
+    $timelc =~ s/,$//;
+    ($sec, $min, $hour, $mday, $mon, $year) = split /,/, $timelc;
+    $year -= 1900;
+    $mon  -= 1;
+    $timelc = timelocal($sec, $min, $hour, $mday, $mon, $year);
+
+    return $timelc;
+}
+
+sub do_power_management_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Power Manangement Tests');
+    select_console('svirt');
+
+    # Guest Power Off
+    record_info('Guest Power Off');
+    check_power_state($vm_id, $vm_ip, 'power.off', 0, $VM_POWERED_ON, $VM_POWERED_OFF);
+
+    # Guest Power On
+    record_info('Guest Power On');
+    check_power_state($vm_id, $vm_ip, 'power.on', 1, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    # Guest Reset
+    record_info('Guest Power Reset');
+    check_power_state($vm_id, $vm_ip, 'power.reset', 1, $VM_POWERED_ON, $VM_POWERED_ON);
+
+    # Guest Restart
+    record_info('Guest Power Restart');
+    check_power_state($vm_id, $vm_ip, 'power.restart', 1, $VM_POWERED_ON, $VM_POWERED_ON);
+
+    # Guest Shutdown
+    record_info('Guest Power Shutdown');
+    check_power_state($vm_id, $vm_ip, 'power.shutdown', 0, $VM_POWERED_ON, $VM_POWERED_OFF);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Networking Tests');
+    check_power_state($vm_id, $vm_ip, 'power.on', 1, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    login_vm_console();
+
+    assert_script_run('ping -c3 openqa.suse.de');
+    assert_script_run('ping -c3 www.suse.com');
```

Comment:
I am not very sure what are you testing here. Because it seems that some services start to provide services with ipv6 address. So your ping may also get a response from ipv6 address. And if your guest has multiple interfaces, it is very difficult to tell the interface on which icmp reply is received. What about make this ping test more specific ? For example, use some options like -4, -I and many others ?

Reply:
Yes, you are right. We should use the option "-4" for ping via ipv4 address. This test is just for testing if assigned ip can work normally. Besides, the guest is designed and assigned only one network interface.